libr/include/r_types_overflow.h

```diff
@@ -0,0 +1,53 @@
+#ifndef R_TYPES_OVERFLOW_H
+#define R_TYPES_OVERFLOW_H
+
+/* preventive math overflow checks */
+#if !defined(SZT_ADD_OVFCHK)
+#define SZT_ADD_OVFCHK(x,y) ((SIZE_MAX - (x)) < (y))
+#endif
+
+#if 0
+// ADD
+if ((x > 0) && (a > INT_MAX - x)) /* `a + x` would overflow */;
+if ((x < 0) && (a < INT_MIN - x)) /* `a + x` would underflow */;
+#endif
+#define UT64_ADD_OVFCHK(x,y) ((UT64_MAX - (x)) < (y))
+#define ST64_ADD_OVFCHK(a,x) ((((x) > 0) && ((a) > ST64_MAX - (x))) || ((x) <0 &&(a)< ST32_MIN - (x)))
```

Comment:
check style, in particular near `... &&(a)...`, here and in next lines.

Reply:
see above